### PR TITLE
Vulnerability: Distinguish between CVSS v2 and v3 score mapping

### PR DESCRIPTION
**Pull request by sschuberth**
See e.g. [1] for the differences.

Also remove the "rating" variable from the Vulnerability class as it is
up to the Advisor implementation now to choose the correct mapping type.

[1] https://www.balbix.com/insights/cvss-v2-vs-cvss-v3/#CVSSv3-Scoring-Scale-vs-CVSSv2-6

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: model/src/main/kotlin/Vulnerability.kt [hunk 1]
```diff
@@ -45,30 +45,52 @@ data class Vulnerability(
     val url: URL? = null
 ) {
     /**
-     * The rating attaches human-readable semantics to the score number, see
-     * https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale.
+     * The rating attaches human-readable semantics to the score number according to CVSS version 2, see
+     * https://www.first.org/cvss/v2/guide.
```
Comment (MarcelBochtler) [inline model/src/main/kotlin/Vulnerability.kt:49]: Can you add a source for the ranges?

File: model/src/main/kotlin/Vulnerability.kt [hunk 2]
```diff
@@ -45,30 +45,52 @@ data class Vulnerability(
...
-            private val bounds = enumValues<Rating>().asIterable().zipWithNext()
+            private val bounds = enumValues<Cvss3Rating>().asIterable().zipWithNext()
             private val limits = listOf(bounds.first().first, bounds.last().second)
 
             /**
-             * Get the [Rating] from a [score], or null if the [score] does not map to any [Rating].
+             * Get the [Cvss3Rating] from a [score], or null if the [score] does not map to any [Cvss3Rating].
              */
-            fun fromCvssScore(score: Float): Rating? {
+            fun fromScore(score: Float): Cvss3Rating? {
```
Comment (oheger-bosch) [inline model/src/main/kotlin/Vulnerability.kt:89]: Both `Cvss2Rating` and `Cvss3Rating` provide `fromScore()` functions, which are semantically equivalent, but use completely different algorithms to calculate the score. This is a bit confusing. Should the implementations be aligned?

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/Vulnerability.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/main/kotlin/Vulnerability.kt:49]: I actually didn't find one on the official CVSS pages for v2 (only for v3 I found the ranges to be explicitly defined). The best source for the ranges (or comparison of ranges) I've found is the link I've mentioned in the commit message. I preferred to not use that link also here, as it's less official, but I could do so if you prefer.

File: model/src/main/kotlin/Vulnerability.kt (hunk 2 above)
Comment (sschuberth) [reply model/src/main/kotlin/Vulnerability.kt:89]: Hmm, I'm not sure I get what you mean here. The implementations *have* to be different as the specs define the mappings differently for v2 and v3. So how could the implementations be aligned?

Comment (oheger-bosch) [reply model/src/main/kotlin/Vulnerability.kt:89]: What I meant is that the implementations could be structurally similar. Currently, one uses a `when` statement while the other iterates over the enum values.

Comment (sschuberth) [reply model/src/main/kotlin/Vulnerability.kt:89]: I actually liked the fact to show off two different ways how these could be implemented, and I usually prefer generic over hard-coded solutions, but in this case I can see your point, and I've added a new commit to align the implementations.

**Review by oheger-bosch — DISMISSED** [review]

File: model/src/main/kotlin/Vulnerability.kt (hunk 1 above)
Comment (MarcelBochtler) [reply model/src/main/kotlin/Vulnerability.kt:49]: Yeah that was exactly the reason why I asked.
IMO it's better to add it here as well, as that source is better than nothing, and future code readers will be less confused where these values come from.

Comment (sschuberth) [reply model/src/main/kotlin/Vulnerability.kt]: Ok, done.

**Review by MarcelBochtler — APPROVED** [review]

**Pull request merged**